modules/caddytls/leaffileloader_test.go

```diff
@@ -0,0 +1,44 @@
+package caddytls
+
+import (
+	"bytes"
+	"context"
+	"encoding/pem"
+	"os"
+	"testing"
+
+	"github.com/caddyserver/caddy/v2"
+)
+
+func TestLeafFileLoader(t *testing.T) {
+	fl := LeafFileLoader{Files: []string{"../../caddytest/caddy.examplecert.pem"}}
+	fl.Provision(caddy.Context{Context: context.Background()})
+
+	out, err := fl.LoadLeafCertificates()
+
+	if err != nil {
+		t.Errorf("Leaf certs file loading test failed: %v", err)
+	}
+	pemBytes := bytes.NewBuffer(nil)
+	pem.Encode(pemBytes, &pem.Block{Type: "CERTIFICATE", Bytes: out[0].Raw})
```

Comment:
Add a length check before using the cert. Also, you can use [`pem.EncodeToMemory`](https://pkg.go.dev/encoding/pem#EncodeToMemory) to avoid having to create the buffer.